sklearnex/cluster/k_means.py

```diff
@@ -12,6 +12,412 @@
+        self._n_iter_ = value
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator.n_iter_ = value
+
+    class BaseKMeans(ABC):
+        def _save_attributes(self):
+            assert hasattr(self, "_onedal_estimator")
+            self.n_features_in_ = self._onedal_estimator.n_features_in_
+            self.fit_status_ = 0
+            self._tol = self._onedal_estimator._tol
+            self._n_init = self._onedal_estimator._n_init
+            self._n_iter_ = self._onedal_estimator.n_iter_
+            self._labels_ = self._onedal_estimator.labels_
+            self._inertia_ = self._onedal_estimator.inertia_
+            self._algorithm = self._onedal_estimator.algorithm
+            self._cluster_centers_ = self._onedal_estimator.cluster_centers_
+            self._sparse = False
+
+            self.n_iter_ = property(get_n_iter, set_n_iter)
+            self.labels_ = property(get_labels, set_labels)
+            self.inertia_ = property(get_labels, set_inertia)
+            self.cluster_centers_ = property(get_cluster_centers, set_cluster_centers)
+
+            self._is_in_fit = True
+            self.n_iter_ = self._n_iter_
+            self.labels_ = self._labels_
+            self.inertia_ = self._inertia_
+            self.cluster_centers_ = self._cluster_centers_
+            self._is_in_fit = False
+
+    @control_n_jobs
+    class KMeans(sklearn_KMeans, BaseKMeans):
+        __doc__ = sklearn_KMeans.__doc__
+        n_iter_, inertia_ = None, None
+        labels_, cluster_centers_ = None, None
+
+        if sklearn_check_version("1.2"):
+            _parameter_constraints: dict = {**sklearn_KMeans._parameter_constraints}
+
+            @_deprecate_positional_args
+            def __init__(
+                self,
+                n_clusters=8,
+                *,
+                init="k-means++",
+                n_init="auto" if sklearn_check_version("1.4") else "warn",
+                max_iter=300,
+                tol=1e-4,
+                verbose=0,
+                random_state=None,
+                copy_x=True,
+                algorithm="lloyd",
+            ):
+                super().__init__(
+                    n_clusters=n_clusters,
+                    init=init,
+                    max_iter=max_iter,
+                    tol=tol,
+                    n_init=n_init,
+                    verbose=verbose,
+                    random_state=random_state,
+                    copy_x=copy_x,
+                    algorithm=algorithm,
+                )
+
+        elif sklearn_check_version("1.0"):
+
+            @_deprecate_positional_args
+            def __init__(
+                self,
+                n_clusters=8,
+                *,
+                init="k-means++",
+                n_init=10,
+                max_iter=300,
+                tol=1e-4,
+                verbose=0,
+                random_state=None,
+                copy_x=True,
+                algorithm="auto",

```

Comment:
based off of a PR I just merged, this should fail for sklearn1.1 for init signature matching. Something unusual is going on in testing.